### PR TITLE
made setErrorMessage prop optional

### PR DESCRIPTION
**Pull request by ivoalmeida**

### CODE REVIEW

File: packages/shared/modals/Selector.tsx [hunk 1]
```diff
@@ -188,19 +188,19 @@ export class SelectorInput extends React.Component<any, SelectorInputState> {
     // If the user deletes an existing inputValue, set isInputValid back to true
     if (inputValue === '') {
       this.setState({ inputValue, isInputValid: true });
-      this.props.setErrorMessage(true);
+      this.props.setErrorMessage && this.props.setErrorMessage(true);
```
Comment (SanjalKatiyar) [inline packages/shared/modals/Selector.tsx:191]: nits:
```suggestion
      this.props.setErrorMessage?.(true);
```

File: packages/shared/modals/Selector.tsx [hunk 2]
```diff
@@ -188,19 +188,19 @@ export class SelectorInput extends React.Component<any, SelectorInputState> {
...
       return;
     }
     const isInputValid = this.isTagValid(inputValue);
-    this.props.setErrorMessage(isInputValid);
+    this.props.setErrorMessage && this.props.setErrorMessage(isInputValid);
```
Comment (SanjalKatiyar) [inline packages/shared/modals/Selector.tsx:195]:
```suggestion
    this.props.setErrorMessage?.(true);
```

File: packages/shared/modals/Selector.tsx [hunk 3]
```diff
@@ -188,19 +188,19 @@ export class SelectorInput extends React.Component<any, SelectorInputState> {
...
     this.setState({ inputValue, isInputValid });
   }
 
   handleChange(tags, changed) {
     // The way we use TagsInput, there should only ever be one new tag in changed
     const newTag = changed[0];
     const isInputValid = this.isTagValid(newTag);
-    this.props.setErrorMessage(isInputValid);
+    this.props.setErrorMessage && this.props.setErrorMessage(isInputValid);
```
Comment (SanjalKatiyar) [inline packages/shared/modals/Selector.tsx:203]:
```suggestion
    tthis.props.setErrorMessage?.(true);
```

**Comment by ivoalmeida** [comment]
> 

@SanjalKatiyar 

This component was not developed by me. It never had any prop types to begin with. Keep this in mind.

I'll add the prop types based on the props I see being used in the component.

**Comment by SanjalKatiyar** [comment]
> > 
> 
> @SanjalKatiyar
> 
> This component was not developed by me. It never had any prop types to begin with. Keep this in mind.
> 
> I'll add the prop types based on the props I see being used in the component.

yes I know !!
Only asking to add as a safety because this component is missing proper types...

File: packages/shared/modals/Selector.tsx [hunk 4]
```diff
@@ -133,13 +133,27 @@ export const selectorToString = (selector: Selector): string => {
...
 
+type SelectorProps = {
+  setErrorMessage?: (isValid: boolean) => void;
+  placeholder?: string;
+  options?: any;
+  autoFocus?: boolean;
+  inputProps?: any;
+  labelClassName?: string;
+  tags?: string[];
+  onChange?: (value: string[]) => void;
```
Comment (SanjalKatiyar) [inline packages/shared/modals/Selector.tsx:144]:
```suggestion
  tags: string[];
  onChange: (value: string[]) => void;
```

**Comment by SanjalKatiyar** [comment]
Just one small change, looks good... thanks for sending quick fix !

**Pull request merged**